SCons/Tool/mslink.py

```diff
@@ -127,12 +127,14 @@ def _dllEmitter(target, source, env, paramtp):
         target[0].attributes.pdb = pdb
 
     if version_num >= 11.0 and env.get('PCH', 0):
-        # MSVC 11 and above need the PCH object file to be added to the link line,
-        # otherwise you get link error LNK2011.
-        pchobj = SCons.Util.splitext(str(env['PCH']))[0] + '.obj'
-        # print "prog_emitter, version %s, appending pchobj %s"%(version_num, pchobj)
-        if pchobj not in extrasources:
-            extrasources.append(pchobj)
+        pch_subst = env.subst("$PCH", target=target, source=source)
+        if pch_subst:
+            # MSVC 11 and above need the PCH object file to be added to the link line,
+            # otherwise you get link error LNK2011.
+            pchobj = SCons.Util.splitext(pch_subst)[0] + '.obj'
```

Comment:
drive by: Should this being using $OBJSUFFIX instead of hardcoding '.obj'?

Reply:
I'd say so.  Just ran into another suffix-hardcoding thing elsewhere the other day, irritating.

Reply:
Yes please!

Reply:
done: 5eb908a